cli/command/cli.go

```diff
@@ -93,12 +96,25 @@ func (cli *DockerCli) Initialize(opts *cliflags.ClientOptions) error {
 	if err != nil {
 		return err
 	}
+
+	cli.defaultVersion = cli.client.ClientVersion()
+
 	if opts.Common.TrustKey == "" {
 		cli.keyFile = filepath.Join(cliconfig.ConfigDir(), cliflags.DefaultTrustKeyFile)
 	} else {
 		cli.keyFile = opts.Common.TrustKey
 	}
 
+	if ping, err := cli.client.Ping(context.Background()); err == nil {
+		cli.hasExperimental = ping.Experimental
+
+		// since the new header was added in 1.25, assume server is 1.24 if header is not present.
+		if ping.APIVersion == "" {
+			ping.APIVersion = "1.24"
```

Comment:
Looks like this isn't working as expected, as I'm no longer able to override the API version manually (using `DOCKER_API_VERSION`;

```
DOCKER_API_VERSION=1.21 ./docker version

Client:
 Version:      1.13.0-dev
 API version:  1.24 (downgraded from 1.21)
 Go version:   go1.7.3
 Git commit:   6d49bbe
 Built:        Tue Nov  8 21:49:11 2016
 OS/Arch:      linux/amd64
Error response from daemon: client is newer than server (client API version: 1.24, server API version: 1.21)
```

Also, this limits the functionality of this PR, as it only automatically downgrades to docker 1.12 and up, and not older versions. Wonder if we need to do something for that (e.g., using the `/version` endpoint to check the actual version from the error-message, but that may be a dirty hack)